filters/cookie/cookie_test.go

```diff
@@ -382,3 +383,101 @@ func TestSetCookie(t *testing.T) {
 		}
 	}
 }
+
+func TestDropCookie(t *testing.T) {
+	for _, tt := range []struct {
+		name        string
+		arg         string
+		cookies     http.Header
+		wantCookies map[string]string
+		wantErr     bool
+	}{
+		{
+			name:        "test no cookies",
+			arg:         "no-cookie",
+			cookies:     nil,
+			wantCookies: nil,
+			wantErr:     false,
+		},
+		{
+			name: "test one cookie not match",
+			arg:  "no-match",
+			cookies: http.Header{
+				"Cookie": []string{
+					"foo=foo1",
+				},
+			},
+			wantCookies: map[string]string{"foo": "foo1"},
+			wantErr:     false,
+		},
+		{
+			name: "test one cookie with match",
+			arg:  "foo",
+			cookies: http.Header{
+				"Cookie": []string{
+					"foo=foo1",
+				},
+			},
+			wantCookies: nil,
+			wantErr:     false,
+		},
+		{
+			name: "test two cookies and one cookie with match",
+			arg:  "foo",
+			cookies: http.Header{
+				"Cookie": []string{
+					"foo=foo1",
```

Comment:
We should add more test cases that contain multiple cookies per value

Reply:
I think there could also be several cookies with the same name

Reply:
done